odoo/addons/base/models/ir_actions_report.py

```diff
@@ -424,6 +424,7 @@ def _run_wkhtmltopdf(
                         'Wkhtmltopdf failed (error code: %s). Memory limit too low or maximum file number of subprocess reached. Message : %s')
                 else:
                     message = _('Wkhtmltopdf failed (error code: %s). Message: %s')
+                _logger.error(message % (str(process.returncode), err[-1000:]))
```

Comment:
Good to have this on log too

Reply:
Considering leaving it up to the logging framework to format the string which makes it lazier, so the string is formatted only when necessary (based on configured log level). Also, formatting the string yourself can have some [undesired side-effects](https://stackoverflow.com/a/13375482/).

I see this is taken from the original `raise UserError(...)`, but I think casting to `str` is redundant here (`%s` should cover us here?).

```suggestion
                _logger.error(message, process.returncode, err[-1000:])
```

Reply:
It llokks like According to the https://hg.python.org/cpython/file/3.6/Lib/logging/__init__.py#l338 it does same internally.
@xmo-odoo  What do you think on this ?

Reply:
@sswapnesh @naglis looks correct on both suggestions. I'd only log at the "warning" level though, error seems too high for something like wkhtmltopdf which will blow up because it got up on the wrong side of the bed this morning.

The interpolation is necessary for the UserError but you could also remove the str() call there as — as @naglis notes — the `%s` in the format string already performs that job.